### ISSUE TITLE
[Event Hubs Client] Enable Test Parallelization

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#5988]
# Summary

The test suite for the the Event Hubs client library currently is unable to run in parallel due to the use of shared state, shared service entities, and code structures that assume a serial environment.  Once these areas have been refactored, it should be possible to run tests in parallel.

# Scope of Work

- Identify tests which rely on running in isolation and cannot be refactored to be safe for concurrent execution.  This will likely be isolated to the diagnostics tests, which rely on infrastructure that emits global information.   

- Tests which rely on isolation should be grouped with those that share the same resources and tagged as a test collection requiring isolated execution.   It is preferable to refactor a test than to include it; all reasonable efforts should be made to minimize tests in these collections.

- The test runner metadata should be updated to denote that test paralleization is allowed.

# Out of Scope

- Refactoring of the general approach for testing, the test cases, code style, and other artifacts of the code base unrelated to timing-related items.

- Adding of additional test cases or new testing scenarios.

# Success Criteria

- Parallelization for test runs should be enabled, covering all tests that were not identified as needing to be part of an isolated collection.

- Tests should continue to produce deterministic and consistent results, with no new failures or added chance of intermittent failures.

- The total test run should complete in 10 minutes or less; if this is not the case, then issues should be filed for investigation of timing issues and speeding up the tests.
